### PR TITLE
Add custom resolvers for dictionary projects

### PR DESCRIPTION
**Pull request by lachlan1m**

### CODE REVIEW

File: packages/query/src/graphql/plugins/PgDictionaryPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      }
+    }
+  }
+};
+
+const PgDictionaryPlugin = makeExtendSchemaPlugin((build, options) => {
+  const [schemaName] = options.pgSchemas;
+  const {pgSql: sql} = build;
+
+  const eventsTableExists = build.pgIntrospectionResultsByKind.class.find(
```
Comment (lachlan1m) [inline packages/query/src/graphql/plugins/PgDictionaryPlugin.ts:57]: [ ] check on table columns as well

File: packages/query/src/graphql/plugins/PgDictionaryPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import {makeExtendSchemaPlugin, gql} from 'graphile-utils';
+
+const SORT_ARGS = {
+  lessThan: '<',
+  lessThanOrEqualTo: '<=',
+  greaterThan: '>',
+  greaterThanOrEqualTo: '>=',
+};
+
+const generateArgs = (queryBuilder: QueryBuilder, args, sql) => {
```
Comment (lachlan1m) [inline packages/query/src/graphql/plugins/PgDictionaryPlugin.ts:14]: - [ ] I can make this cleaner

Comment (ianhe8x) [reply packages/query/src/graphql/plugins/PgDictionaryPlugin.ts:14]: You can use `builder.hook("GraphQLObjectType:fields:field:args",` like in here https://github.com/graphile-contrib/postgraphile-plugin-connection-filter/blob/fda1bc1e4474a3a64be10f3228c81b88560f6799/src/PgConnectionArgFilterPlugin.ts#L27

File: packages/query/src/graphql/plugins/PgDictionaryPlugin.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        distinctExtrinics(on: String!): ExtrinsicsConnection!
+        distinctSpecVersions(on: String!): SpecVersionsConnection!
+      }
+    `,
+    resolvers: {
+      Query: {
+        distinctEvents: async (_parentObject, args, _context, info): Promise<any> => {
+          if (eventsTableExists) {
+            return info.graphile.selectGraphQLResultFromTable(
+              sql.fragment`(select distinct on (${sql.identifier(args.on)}) * from ${sql.identifier(
```
Comment (lachlan1m) [inline packages/query/src/graphql/plugins/PgDictionaryPlugin.ts:36]: sql needs to be updated

**Pull request closed without merging**